src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4342,6 +4341,43 @@ BoundExpression bindObjectCreationExpression(ObjectCreationExpressionSyntax node
             }
         }
 
+        private BoundExpression BindCollectionCreationExpression(CollectionCreationExpressionSyntax syntax, BindingDiagnosticBag diagnostics)
+        {
+            MessageID.IDS_FeatureCollectionLiterals.CheckFeatureAvailability(diagnostics, syntax, syntax.OpenBracketToken.GetLocation());
+
+            var builder = ArrayBuilder<BoundExpression>.GetInstance(syntax.Elements.Count);
+            foreach (var element in syntax.Elements)
+            {
+                builder.Add(bindElement(element, diagnostics));
+            }
+            return new BoundUnconvertedCollectionLiteralExpression(syntax, builder.ToImmutableAndFree(), this);
+
+            BoundExpression bindElement(CollectionElementSyntax syntax, BindingDiagnosticBag diagnostics)
+            {
+                switch (syntax)
+                {
+                    case ExpressionElementSyntax expressionElementSyntax:
+                        return BindValue(expressionElementSyntax.Expression, diagnostics, BindValueKind.RValue);
+
+                    case DictionaryElementSyntax dictionaryElementSyntax:
+                        _ = BindValue(dictionaryElementSyntax.KeyExpression, diagnostics, BindValueKind.RValue);
+                        _ = BindValue(dictionaryElementSyntax.ValueExpression, diagnostics, BindValueKind.RValue);
+                        // PROTOTYPE: Temporary error until feature is implemented.
+                        Error(diagnostics, ErrorCode.ERR_InvalidExprTerm, syntax, syntax);
+                        return BadExpression(syntax);
+
+                    case SpreadElementSyntax spreadElementSyntax:
+                        _ = BindValue(spreadElementSyntax.Expression, diagnostics, BindValueKind.RValue);
+                        // PROTOTYPE: Temporary error until feature is implemented.
+                        Error(diagnostics, ErrorCode.ERR_InvalidExprTerm, syntax, syntax);
+                        return BadExpression(syntax);
```

Comment:
nit: consider specific errors here to make it clear that these just aren't supported currently.  saying that it's an invalid-expr seems like it could be confusing for anyone trying this out.